frontend/packages/pipelines-plugin/src/components/repository/hooks/pac-hook.ts

```diff
@@ -1,8 +1,28 @@
+import * as React from 'react';
 import { useK8sGet } from '@console/internal/components/utils/k8s-get-hook';
 import { ConfigMapModel } from '@console/internal/models';
-import { ConfigMapKind } from '@console/internal/module/k8s';
+import { ConfigMapKind, k8sList } from '@console/internal/module/k8s';
+import { RepositoryModel } from '@console/pipelines-plugin/src/models';
 import { PAC_INFO } from '../../pac/const';
 import { PIPELINE_NAMESPACE } from '../../pipelines/const';
 
 export const usePacInfo = () =>
   useK8sGet<ConfigMapKind>(ConfigMapModel, PAC_INFO, PIPELINE_NAMESPACE);
+
+export const useRepositoryPresent = (repoURL) => {
+  const [repoAlreadyExists, setRepoAlreadyExists] = React.useState(false);
+
+  React.useEffect(() => {
+    // eslint-disable-next-line promise/catch-or-return
+    k8sList(RepositoryModel).then((repos) => {
+      if (repos && repos.length > 0) {
+        const repo = repos.find((r) => r.spec.url === repoURL);
+        if (repo) {
+          setRepoAlreadyExists(true);
+        }
+      }
```

Comment:
Shouldn't this also jump back to false if you change the `baseURL`?

```suggestion
      setRepoAlreadyExists(repos.some((r) => r.spec.url === repoURL));
```